apps/frontend/core-dapp/src/stores/TokenSenderStore.ts

```diff
@@ -0,0 +1,41 @@
+import { ContractReturn, ContractStore, Factory } from 'prepo-stores'
+import { BigNumber } from 'ethers'
+import { RootStore } from './RootStore'
+import { TokenSenderAbi, TokenSenderAbi__factory } from '../../generated/typechain'
+import { SupportedContracts } from '../lib/contract.types'
+
+type GetScaledPrice = TokenSenderAbi['functions']['getScaledPrice']
+type GetScaledPriceLowerBound = TokenSenderAbi['functions']['getScaledPriceLowerBound']
+export class TokenSenderStore extends ContractStore<RootStore, SupportedContracts> {
+  constructor(rootStore: RootStore) {
+    super(rootStore, 'TOKEN_SENDER', TokenSenderAbi__factory as unknown as Factory)
+  }
+
+  private getScaledPrice(...params: Parameters<GetScaledPrice>): ContractReturn<GetScaledPrice> {
+    return this.call<GetScaledPrice>('getScaledPrice', params)
+  }
+
+  private getScaledPriceLowerBound(
+    ...params: Parameters<GetScaledPriceLowerBound>
+  ): ContractReturn<GetScaledPriceLowerBound> {
+    return this.call<GetScaledPriceLowerBound>('getScaledPriceLowerBound', params)
+  }
+
+  private get scaledPriceIfActive(): BigNumber | undefined {
+    const scaledPrice = this.getScaledPrice()?.[0]
+    const scaledPriceLowerBound = this.getScaledPriceLowerBound()?.[0]
+
+    if (scaledPrice === undefined || scaledPriceLowerBound === undefined) return undefined
+
+    return scaledPrice.lte(scaledPriceLowerBound) ? undefined : scaledPrice
```

Comment:
I'm thinking we should use undefined to indicate loading and 'zero' value to indicate there's no reward and that we should hide the reward text when there's no reward. So I'd probably return 0 here 

```suggestion
    return scaledPrice.lte(scaledPriceLowerBound) ? BigNumber.from(0) : scaledPrice
```